File: app/src/main/java/com/tresor/home/activity/AddPaymentActivity.java
package com.tresor.home.activity;

import android.app.Activity;

import com.tresor.common.activity.addpayment.PaymentTemplate;
import com.tresor.common.model.viewmodel.SpendingModel;
import com.tresor.home.inteface.HomeActivityListener;
import com.tresor.home.model.FinancialHistoryModel;
import com.tresor.home.model.SpendingModelWrapper;

import java.util.ArrayList;

/**
 * Created by kris on 10/10/17. Tokopedia
 */

public class AddPaymentActivity extends PaymentTemplate {
    @Override
    protected SpendingModelWrapper initialModel() {
        return new SpendingModelWrapper(0, generatePlainSpendingModel());
    }

    @Override
    protected int imageChoosen() {
        return 0;
    }

    @Override
    protected int getMode() {
        return HomeActivityListener.ADD_NEW_PAYMENT_REQUEST_CODE;
    }

    private SpendingModel generatePlainSpendingModel() {
        return new SpendingModel(0,
                "0",
                0,
                false,
                1,
                "",
                "",
                0,
                new ArrayList<String>(),
                "");
    }
}

File: app/src/main/java/com/tresor/home/activity/AddPaymentActivity.kt
package com.tresor.home.activity

import android.content.Context
import android.content.Intent
import com.tresor.common.activity.addpayment.PaymentTemplateKotlin
import com.tresor.common.model.viewmodel.SpendingModel
import com.tresor.home.inteface.HomeActivityListener
import com.tresor.home.model.SpendingModelWrapper

import java.util.ArrayList

/**
 * Created by kris on 10/10/17. Tokopedia
 */

fun Context.addPaymentActivityIntent(): Intent {
    return Intent(this, AddPaymentActivity::class.java)
}

class AddPaymentActivity : PaymentTemplateKotlin() {
    override fun imageChosen(): Int {
        return INITIAL_IMAGE_SELECTOR_POSITION
    }

    val INITIAL_ITEM_POSITION = 0
    val INITIAL_IMAGE_SELECTOR_POSITION = 0

    override fun initialModel(): SpendingModelWrapper {
        return SpendingModelWrapper(INITIAL_ITEM_POSITION, generatePlainSpendingModel())
    }

    override fun getMode(): Int {
        return HomeActivityListener.ADD_NEW_PAYMENT_REQUEST_CODE
    }

    private fun generatePlainSpendingModel(): SpendingModel {
        return SpendingModel(
                0,
                "0",
                0.0,
                false,
                1,
                "",
                "",
                0,
                ArrayList(),
                ""
        )
    }
}
